staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/queueset/queueset_test.go

```diff
@@ -32,6 +34,70 @@ import (
 	"k8s.io/klog/v2"
 )
 
+type SortFns struct {
+	LenFn  func() int
+	LessFn func(i, j int) bool
+	SwapFn func(i, j int)
+}
+
+func (sc *SortFns) Len() int           { return sc.LenFn() }
+func (sc *SortFns) Less(i, j int) bool { return sc.LessFn(i, j) }
+func (sc *SortFns) Swap(i, j int)      { sc.SwapFn(i, j) }
+
+// FairAlloc computes the min-max fair allocation of the given
+// capacity to the given demands (which slice is not side-effected).
+func FairAlloc(demands []float64, capacity float64) []float64 {
+	lim := len(demands)
+	indices := make([]int, lim)
+	for i := 0; i < lim; i++ {
+		indices[i] = i
+	}
+	sort.Sort(&SortFns{
+		LenFn:  func() int { return lim },
+		LessFn: func(i, j int) bool { return demands[indices[i]] < demands[indices[j]] },
```

Comment:
Please use https://golang.org/pkg/sort/#Slice instead?

Reply:
Or, more directly, `sort.Float64s`.  That's where I started.  Then I realized that my problem is not just sorting the input --- the output slice needs to be correlated with the input.  So that's why this code creates an `indices` slice to use for indexing both input and output.

Reply:
Yes, you can't use sort.Float64s, but you can use sort.Slice (on the indicies slice, not the demands slice).

Reply:
Oh, I see.  Yes.

Reply:
done